core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -1007,14 +1014,27 @@ private void reconcileOpenPartitions(long prevPartitionVersion) {
         reconcileOpenPartitionsFrom(0);
     }
 
+    private void releaseTxn() {
+        if (txnAcquired) {
+            txnScoreboard.releaseTxn(txn);
+            txnAcquired = false;
+        }
+    }
+
     private boolean reload(boolean activation) {
         if (this.txn == txFile.unsafeReadTxn()) {
             if (activation) {
-                txnScoreboard.acquireTxn(txn);
+                acquireTxn();
             }
             return false;
         }
-        return reloadSlow();
+        try {
+            reloadSlow();
```

Comment:
minor: there is a way now to call `reload()` on passive reader and it will acquire txn while being passive. I think we should check that `reload()` is never called on a passive reader.

Reply:
It turns out out of pool `TableReader` instance is "passive"